wagtail/search/backends/database/sqlite/query.py

```diff
@@ -0,0 +1,239 @@
+    # it's consistent with other usage in Django.
+    def bitand(self, other):
+        return self._combine(other, self.BITAND, False)
+
+    def bitor(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __or__(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __and__(self, other):
+        return self._combine(other, self.BITAND, False)
+
+
+class SearchQueryField(Field):
+
+    def db_type(self, connection):
+        return None
+
+
+class Lexeme(LexemeCombinable, Value):
+    _output_field = SearchQueryField()
+
+    def __init__(self, value, output_field=None, *, prefix=False, weight=None):
+        self.prefix = prefix
+        self.weight = weight
+        super().__init__(value, output_field=output_field)
+
+    def as_sql(self, compiler, connection):
+        param = "%s" % self.value.replace("'", "''").replace("\\", "\\\\")
+
+        template = "\"%s\""
+
+        label = ''
+        if self.prefix:
+            label += '*'
+
+        if label:
+            param = '{}{}'.format(param, label)
+
+        return template, [param]
+
+
+class CombinedLexeme(LexemeCombinable):
+    _output_field = SearchQueryField()
+
+    def __init__(self, lhs, connector, rhs, output_field=None):
+        super().__init__(output_field=output_field)
+        self.connector = connector
+        self.lhs = lhs
+        self.rhs = rhs
+
+    def as_sql(self, compiler, connection):
+        value_params = []
+        lsql, params = compiler.compile(self.lhs)
+        value_params.extend(params)
+
+        rsql, params = compiler.compile(self.rhs)
+        value_params.extend(params)
+
+        combined_sql = '{} {} {}'.format(lsql, self.connector, rsql)
+        combined_value = combined_sql % tuple(value_params)
+        return '%s', [combined_value]
+
+
+class SearchQueryCombinable:
+    BITAND = 'AND'
+    BITOR = 'OR'
+
+    def _combine(self, other, connector: str, reversed: bool = False):
+        if not isinstance(other, SearchQueryCombinable):
+            raise TypeError(
+                'SearchQuery can only be combined with other SearchQuery '
+                'instances, got %s.' % type(other).__name__
+            )
+        if reversed:
+            return CombinedSearchQuery(other, connector, self)
+        return CombinedSearchQuery(self, connector, other)
+
+    # On Combinable, these are not implemented to reduce confusion with Q. In
+    # this case we are actually (ab)using them to do logical combination so
+    # it's consistent with other usage in Django.
+    def __or__(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __ror__(self, other):
+        return self._combine(other, self.BITOR, True)
+
+    def __and__(self, other):
+        return self._combine(other, self.BITAND, False)
+
+    def __rand__(self, other):
+        return self._combine(other, self.BITAND, True)
+
+
+class SearchQuery(SearchQueryCombinable, Expression):
```

Comment:
To resolve https://github.com/wagtail/wagtail/pull/7420/files#r693936913, perhaps we could rename this to ``SearchQueryExpression`` so that the name no longer clashes with ``wagtail.search.query.SearchQuery``, what do you think?